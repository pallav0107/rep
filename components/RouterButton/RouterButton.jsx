import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Button } from '@wdpui/react-gel';
import { withRouter } from 'react-router-dom';

class RouterButton extends Component {
  static propTypes = {
    history: PropTypes.shape({}).isRequired,
    to: PropTypes.string,
  };

  static defaultProps = {
    to: '/',
  };

  handleClick = () => {
    const { history, to } = this.props;
    history.push(to);
  };

  render() {
    return <Button onClick={this.handleClick} {...this.props} />;
  }
}

export default withRouter(RouterButton);
