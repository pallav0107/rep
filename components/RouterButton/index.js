export { default as RouterButton } from './RouterButton';
