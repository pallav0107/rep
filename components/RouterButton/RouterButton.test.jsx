import React from 'react';
import { shallow } from 'enzyme';
import { BrowserRouter } from 'react-router-dom';
import RouterButton from './RouterButton';
import "../../setupTests"

describe('CustomerProfile', () => {
  it('should render', () => {
    const component = (
      <BrowserRouter>
        <RouterButton />
      </BrowserRouter>
    );
    const wrapper = shallow(component).dive();
    expect(wrapper).toMatchSnapshot();
  });
});
