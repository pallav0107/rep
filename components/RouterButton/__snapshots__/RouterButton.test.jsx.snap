// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomerProfile should render 1`] = `<withRouter(RouterButton) />`;
